ide/coqide/wg_MessageView.ml

```diff
@@ -108,17 +177,20 @@ let message_view () : message_view =
         last_width <- width;
         buffer#set_text "";
         buffer#move_mark (`MARK mark) ~where:buffer#start_iter;
-        List.(iter insert_msg (rev msgs))
+        List.(iter insert_msg (rev !msgs))
       end
 
     method clear =
-      msgs <- []; self#refresh true
+      msgs := []; self#refresh true
 
     method push level msg =
-      msgs <- (level, msg) :: msgs;
+      msgs := (level, msg) :: !msgs;
       insert_msg (level, msg);
       push#call (level, msg)
 
+    method debug_prompt msg =
+      insert_ltac_debug_prompt msg
```

Comment:
Eventually the debugger I/O should move to its own window I understand? So adding a flag to mark something as a prompt seems fine to me for, will do. However there is still a conceptual problem in the sense of what it does mean for that list to contain two prompts?

Reply:
In the sense that a prompt is not only a newline, but also it actives edit mode on the window!

Reply:
After considering it more, it would be cleaner to fix the extra newline by
- having `Comm.output` and `Comm.goal` append `fnl ()` to the message (done once in `tactic_debug.ml` and
- removing the line in `insert_msg` that appends a newline.
Then a 3-tuple is not needed (which is a bit ugly).

> However there is still a conceptual problem in the sense of what it does mean for that list to contain two prompts? ... In the sense that a prompt is not only a newline, but also it actives edit mode on the window!

Detaching the messages window doesn't call `insert_ltac_debug_prompt`, doesn't change edit mode, so not an issue.

> Eventually the debugger I/O should move to its own window I understand?

Still thinking that through.  FWIW I'd like to give the debugger info more structure, e.g. a panel for the stack (click on a line to jump to the code), a panel for variables and a messages/terminal panel.

Reply:
I've opted to keep the list of messages for now, but indeed I am pretty sure you will like to a custom UI with stacks, etc... I think the protocol / code is in good shape to make this reasonably easy.